cpr/accept_encoding.cpp

```diff
@@ -0,0 +1,26 @@
+#include "cpr/accept_encoding.h"
+#include <algorithm>
+#include <numeric>
+
+namespace cpr {
+
+AcceptEncoding::AcceptEncoding(const std::initializer_list<AcceptEncodingMethods>& enum_methods) {
+    methods_.clear();
+    std::transform(enum_methods.begin(), enum_methods.end(), std::back_inserter(methods_),
```

Comment:
`std::transform` is too new. It got introduced in cpp17.

Reply:
I have investigated `std::accumulate` and `std::transform` already; however, the cppcheck throws the error messages to advise me to use that two methods. 
Could you give me some suggestions?

Reply:
BTW, `std::transform` is used in curl_container.cpp

Reply:
I tried another way to pass Cppcheck: wrap the raw loop into another function and then make them more readable. 
Reference: https://belaycpp.com/2021/06/22/dont-use-raw-loops/

Reply:
Turns out I was wrong there. They both are available in cpp11. They just changed the signature in later versions and extended it. Sorry for the confusion. This means it can stay how it is.

Reply:
OK, I would like to keep it with `std::accumulate` and `std::transform` for simplicity.